lib/rules/no-invalid-position-declaration/index.mjs

```diff
@@ -0,0 +1,57 @@
+import findNodeUpToRoot from '../../utils/findNodeUpToRoot.mjs';
+import { nestingSupportedAtKeywords } from '../../reference/atKeywords.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'no-invalid-position-declaration';
+
+const messages = ruleMessages(ruleName, {
+	rejected: 'Unexpected invalid position declaration',
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/no-invalid-position-declaration',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			const { parent } = decl;
```

Comment:
```suggestion
		root.walkDecls((decl) => {
			if (!isStandardSyntaxDeclaration(decl)) return;
			
			const { parent } = decl;
```

Let's ignore non-standard declarations, such as `$foo`, so that SCSS users can use the rule.